Dockerfile

```diff
@@ -58,21 +64,30 @@ ENV BUILD_ENV=${BUILD_ENV}
 # server (Gunicorn). Heroku will ignore this.
 EXPOSE 8000
 
+# Install poetry using the installer (keeps Poetry's dependencies isolated from the app's)
+# chown protects us against cases where files downloaded by poetry have invalid ownership
+# chmod ensures poetry dependencies are accessible when packages are installed
+RUN curl -sSL https://install.python-poetry.org | python3 -
+RUN chown -R root:root ${POETRY_HOME} && \
+    chmod -R 0755 ${POETRY_HOME}
+
 # Don't use the root user as it's an anti-pattern and Heroku does not run
 # containers as root either.
 # https://devcenter.heroku.com/articles/container-registry-and-runtime#dockerfile-commands-and-runtime
 USER guide
 
 # Install your app's Python requirements.
 RUN python -m venv $VIRTUAL_ENV
-COPY --chown=guide ./requirements/ ./requirements/
-RUN pip install --upgrade pip && pip install -r ./requirements/production.txt
+COPY --chown=guide pyproject.toml poetry.lock ./
+RUN pip install --upgrade pip && poetry install ${POETRY_INSTALL_ARGS} --no-root
 
 COPY --chown=guide --from=frontend ./apps/frontend/static ./apps/frontend/static
 
 # Copy application code.
 COPY --chown=guide . .
 
+RUN poetry install ${POETRY_INSTALL_ARGS}
```

Comment:
I haven’t tested the Dockerfile changes locally, @laymonage will leave it for you to to troubleshoot if the deployment fails.